UnicodeJsps/src/main/java/org/unicode/jsp/UBAVersion.java

```diff
@@ -0,0 +1,72 @@
+package org.unicode.jsp;
+
+import org.unicode.props.UcdPropertyValues.Age_Values;
+
+/**
+ * A class to encapsulate the available C UBA versions.
```

Comment:
This looks ok, but why not define a Java enum?
You could still have custom functions for where the defaults are inconvenient.

Reply:
I thought to make it a custom object because it references the UcdPropertyValues.Age_Values — but it could not.

Reply:
I guess I could have simply `V11_0, V12_3` etc. and then change `_` -> `.` where needed.

Reply:
@markusicu Is it a valid assumption that there will be a org.unicode.props.UcdPropertyValues.Age_Values for each UBA version?

Reply:
It is valid that there will be a Unicode Age corresponding to each UBA version, but not vice versa -- at least not from the point of view of the bidic code, since the reference code only covers versions starting from 6.2. (And it might conceivably skip versions -- I've just been adding each version to the reference code as I update it.)

Reply:
Hmm, this code is actually really close to being able to convert to a [Java enum](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html) (I think you could get rid of a little code if you do because enums also maintain an ordering of defined values via an array than you get via `.values()`). 

FWIW, `Age_Values` is already [defined as an enum](https://github.com/unicode-org/unicodetools/blob/main/unicodetools/src/main/java/org/unicode/props/UcdPropertyValues.java#L34).

Reply:
Right. This is just for bidic.jsp, which needs
- getCurrent() (=get latest)
- iterate over all values
- for each value: get the Age_Values.getShortName()
- for each value: same, but without dots

This could be a new enum type with an Age_Values field, constructing each UBAVersion with an Age_Values.

Or, this could simply be an EnumSet<Age_Values>.of(... subset of Age_Values ...), pulling the tiny bit of custom logic into bidic.jsp. No real need for a new type.

Reply:
If we need this drop down.. can you give a brief snippet of how you expect the enum to look?

Reply:
If, yes. I don't have an opinion on whether we need it :-)

I would just put something like this into bidic.jsp:
```Java
private static final EnumSet<Age_Values> UBA_VERSIONS = EnumSet.of(V6_2, V6_3, ...);
```
And then just iterate over the set, fetch the last one from iteration or from toArray()/size(), call age.getShortName(), and get the three-digit version from an inline expression.

Reply:
> If, yes. I don't have an opinion on whether we need it :-)
> 
> I would just put something like this into bidic.jsp:
> 
> ```java
> private static final EnumSet<Age_Values> UBA_VERSIONS = EnumSet.of(V6_2, V6_3, ...);
> ```
> 
> And then just iterate over the set, fetch the last one from iteration or from toArray()/size(), call age.getShortName(), and get the three-digit version from an inline expression.

oh, an enumset of `Age_Values` … not a new enum.  I was struggling to see how defining the identity of the new enum's values would be a benefit.

Reply:
> oh, an enumset of `Age_Values` … not a new enum. I was struggling to see how defining the identity of the new enum's values would be a benefit.

Well, a new enum is a possibility, and is what I initially thought of. What you built is basically a poor man's enum. By turning it into a real enum, you could remove some boilerplate, such as (as Elango wrote) using `.values()`. But there is already an enum, and what you need is a subset of its values, so an EnumSet seems good enough. And if it's used only in one place, we need not define special API.

Reply:
I would rather not put more code into the JSP. JSPs are excruciatingly error prone and unmaintainable. But I can put the relevant code into statics on UBAVersion, and see what you think of that.

Reply:
OK @echeran @macchiati @markusicu  please take a look… it's now an enumset.